prover/lang/kore-lang.md

```diff
@@ -461,60 +461,47 @@ and values, passed to K's substitute.
        => SetItem(N) PatternsToVariableNameSet(Ps)
 ```
 
-Substitution: Substitute term or variable
+Capture-free substitution: Substitute term or variable.
+
+TODO: This allows us to substitute arbitary terms (and not just variables) for
+terms. This is very non-standard. This is currently needed because the various
+unfolding strategies use this function. They should instead generate a context,
+where the term being unfolded has been replace by `#hole`.
 
 ```k
   syntax PatternOrVarName ::= Pattern | VariableName
-
-  // capture-avoiding substitution
-  syntax Pattern ::= casubst(Pattern, PatternOrVarName, Pattern)    [function]
-
-  rule casubst(P, V, S) => #subst(true, P, V, S)
-
-  // non-capture-avoiding substitution
   syntax Pattern ::= subst(Pattern, PatternOrVarName, Pattern)    [function, klabel(subst)]
```

Comment:
So `subst` only works for free variables, but it is not the capture-avoiding substitution. Is that right?

Reply:
No, This commit removes the old `subst` and renames `casubst` to `subst`. We don't need non-capture avoiding substitution.